### PR TITLE
DropDownEditor: add overlay class to wrapper when popup is opening (T1118164)

### PR DESCRIPTION
**Pull request by marker-dao**

### CODE REVIEW

**Comment by ksercs** [comment]
What about tests?
PR name is confusing. Fix is not on ddBox level

File: js/ui/drop_down_editor/ui.drop_down_editor.js [hunk 1]
```diff
@@ -212,9 +213,30 @@ const DropDownEditor = TextBox.inherit({
...
         this._options.cache('dropDownOptions', this.option('dropDownOptions'));
     },
 
+    _mergeDropDownOptions: function() {
+        const options = this.option('dropDownOptions');
+        const defaultConfig = this._popupConfig();
+        const wrapperClasses = this._mergeWrapperClasses(defaultConfig, options);
+
+        if(options.wrapperAttr?.class) {
+            options.wrapperAttr.class = wrapperClasses;
```
Comment (ksercs) [inline js/ui/drop_down_editor/ui.drop_down_editor.js]: I think it's not a good idea to mutate public option's value

Comment (marker-dao) [reply js/ui/drop_down_editor/ui.drop_down_editor.js]: We can't add this option by another way.

File: js/ui/drop_down_editor/ui.drop_down_editor.js [hunk 2]
```diff
@@ -636,7 +637,9 @@ const DropDownEditor = TextBox.inherit({
         }
     },
 
-    _popupShowingHandler: noop,
+    _popupShowingHandler() {
+        $(this._popup.$wrapper()).addClass(DROP_DOWN_EDITOR_OVERLAY);
```
Comment (ksercs) [inline js/ui/drop_down_editor/ui.drop_down_editor.js]: can `this._popup.$wrapper()` really be not a jquery element? if no, let's get rid of excess wrap

File: js/ui/html_editor/ui/formDialog.js [hunk 3]
```diff
@@ -66,7 +66,7 @@ class FormDialog {
             deferRendering: false,
             focusStateEnabled: false,
             showCloseButton: false,
-            wrapperAttr: { class: 'dx-formdialog' },
+            wrapperAttr: { class: DIALOG_CLASS },
```
Comment (ksercs) [inline js/ui/html_editor/ui/formDialog.js]: why do we still use `wrapperAttr` here if it's a reason of initial problem?

File: testing/tests/DevExpress.ui.widgets.editors/dropDownEditor.tests.js [hunk 4]
```diff
@@ -1183,6 +1183,50 @@ QUnit.module('Templates', () => {
     }
 });
 
+QUnit.module('Drop Down Popup', {
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/dropDownEditor.tests.js]: Why do we need separate high-level module for this tests? We already have `popup integration` module. Let's move this module there, name it `ios tests` and move tests' common code to beforeEach

File: testing/tests/DevExpress.ui.widgets/dropDownButton.tests.js [hunk 5]
```diff
@@ -1116,6 +1117,49 @@ QUnit.module('list integration', {}, () => {
     });
 });
 
+QUnit.module('Drop Down Popup', {
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets/dropDownButton.tests.js]: here is the same as in ddEditor.tests

File: js/ui/drop_down_editor/ui.drop_down_editor.js (hunk 2 above)
Comment (marker-dao) [reply js/ui/drop_down_editor/ui.drop_down_editor.js]: https://github.com/DevExpress/DevExtreme/pull/22834/commits/4c923621f7e62f19e104b944605080232aa61c93

File: testing/tests/DevExpress.ui.widgets.editors/dropDownEditor.tests.js (hunk 4 above)
Comment (marker-dao) [reply testing/tests/DevExpress.ui.widgets.editors/dropDownEditor.tests.js]: https://github.com/DevExpress/DevExtreme/pull/22834/commits/f94bb40f05bacbe31caa6768f9e639e701d615da

File: testing/tests/DevExpress.ui.widgets/dropDownButton.tests.js (hunk 5 above)
Comment (marker-dao) [reply testing/tests/DevExpress.ui.widgets/dropDownButton.tests.js]: https://github.com/DevExpress/DevExtreme/pull/22834/commits/f94bb40f05bacbe31caa6768f9e639e701d615da

File: js/ui/html_editor/ui/formDialog.js (hunk 3 above)
Comment (marker-dao) [reply js/ui/html_editor/ui/formDialog.js]: https://github.com/DevExpress/DevExtreme/pull/22834/commits/14aaa27b61571215ec40b31e0ba8fac2741eb32b

File: js/ui/drop_down_editor/ui.drop_down_editor.js [hunk 6]
```diff
@@ -594,7 +594,8 @@ const DropDownEditor = TextBox.inherit({
             toolbarItems: this._getPopupToolbarItems(),
             onPositioned: this._popupPositionedHandler.bind(this),
             fullScreen: false,
-            contentTemplate: null
+            contentTemplate: null,
+            _wrapperClassExternal: DROP_DOWN_EDITOR_OVERLAY,
```
Comment (ksercs) [inline js/ui/drop_down_editor/ui.drop_down_editor.js:598]: what about test for this?

Comment (marker-dao) [reply js/ui/drop_down_editor/ui.drop_down_editor.js:598]: You can find them here: https://github.com/DevExpress/DevExtreme/pull/22834/files/77251c7cbd76f6335cf6309d9179f06f596cbba7#diff-36f1cbaf05bec6a0c2979ea420d3869320b6dbd5657a5d32204fa9551ff8fcb7R1715

**Review by ksercs — APPROVED** [review]

**Review by EugeniyKiyashko — APPROVED** [review]

**Pull request merged**